bin/pycbc_multi_inspiral

```diff
@@ -136,20 +144,79 @@ opt.insert_optimization_option_group(parser)
 inject.insert_injfilterrejector_option_group_multi_ifo(parser)
 sgchisq.SingleDetSGChisq.insert_option_group(parser)
 args = parser.parse_args()
-init_logging(args.verbose)
+init_logging(args.verbose) # takes the --verbose option -> (bool)
 # Set GRB time variable for convenience
-t_gps = args.trigger_time
+t_gps = args.trigger_time # takes the --trigger-time option -> (float)
+# Put the ifos in alphabetical order so they are always called in
+# the same order. Takes the --instruments option -> (list of str)
 # Put the ifos in alphabetical order so they are always called in
 # the same order.
 args.instruments.sort()
-strain.verify_strain_options_multi_ifo(args, parser, args.instruments)
+nifo = len(args.instruments[:])
+# The following class verification methods are used to check whether input CLI
+# options provided by parser to pycbc.strain, pycbc.strain.StrainSegments,
+# pycbc.psd, pycbc.scheme & pycbc.fft modules are sane.
+# pycbc.strain CLI verifier:
+#    the class methods "verify_strain_options_multi_ifo" &
+#                      "verify_segment_options_multi_ifo"
+#    both rely on pycbc.types "required_opts_multi_ifo" to check the CLI options
+# Both methods take the --instruments option -> (list of str)
+strain.verify_strain_options_multi_ifo(
+                                      args, parser, args.instruments)
 strain.StrainSegments.verify_segment_options_multi_ifo(
-        args, parser, args.instruments)
-psd.verify_psd_options_multi_ifo(args, parser, args.instruments)
+                                      args, parser, args.instruments)
+# The pycbc.psd CLI verifier:
+#       relies on:
+#	   ensure_one_opt_multi_ifo -> ensures only options for one
+#                                      detector are defined at a time.
+#          required_opts_multi_ifo  -> used only if psd_estimation is turned on.
+#                                      Checks that CLI options are sane.
+# The following method takes the --instruments option -> (list of str)
+psd.verify_psd_options_multi_ifo(
+                                args, parser, args.instruments)
+# The pycbc.scheme CLI verifier:
+#       these options are used below by the context manager "ctx"
+#       to properly manage resources without memory leaks
+# This method takes as input the option
+# --processing-scheme as name:N -> (str:int)
+# where name can be either mkl, cpu.
+# N is the number of threads(mkl) or CPUs(cpu)
+# NOTE:--processing-scheme can also be set as cuda for GPU usage.
+# However, the cuda scheme breaks the interaction with lalframe.
```

Comment:
I am not sure how useful it is to spell out the various CLI options as comments here (and in the similar comments just below). What if someone changes one option, will they remember to come here and update the comment as well? Also, the argparse code should be fairly self-explanatory to anyone who has a basic understanding of Python and is willing to dig a bit into the PyCBC code base to see what the various `verify_*options()` functions do. If I wanted to comment this part of the code, I would probably write something like "check the CLI options pertaining to computing optimizations", "check the CLI options controlling how FFTs are executed", etc. But again, this is not really where comments are important.

Reply:
I agree. Following that line of thoughts, I think two things are yet to be done:

Reply:
I also noticed that the current version does not have parser.add_argument("--processing-scheme"), Then I wonder how is this being parsed... 

Such parser argument should have the following in the help kwarg:

help = " takes as input the processing method "
           " as name:N.  Where name can be either"
           " mkl or cpu, and N is the number of threads"
           " It can be set to cuda for GPU processing"

Reply:
Moreover, I noticed a few weeks ago that certain combinations of  --trig-end-time,  --trig-start-time, --segment-length,  --segment-start-pad, --segment-end-pad, may make the code complain at the pycbc.strain level. I will open an Issue soon, because we may need to avoid this with assert statements or something similar. 

To be more explicit, with the current version of the multi_inspiral example in https://github.com/gwastro/pycbc/blob/master/examples/multi_inspiral/run.sh , setting the option --segment-length 128 breaks the example. 

I've checked why this is happens and turns out that certain combinations of --segment-start-pad, --segment-end-pad & --segment-length cause the segment_width computed in:
master/pycbc/strain/strain.py to have values <=0

Reply:
@sebastiangomezlopez it seems you have addressed my comment here, and you opened separated issues for your other comments (https://github.com/gwastro/pycbc/issues/4553 and https://github.com/gwastro/pycbc/issues/4554), so I am marking this as resolved.